jpw1991/chebs-necromancy
# 2.5.9: fix a null object exception for structure friendly-fire code; …

# Description

- fix a null object exception for structure friendly-fire code
- ships ignore minion impact damage
- fix duplicate resources on death

Compiled pre-release version [here](https://github.com/jpw1991/chebs-necromancy/releases/tag/2.5.9).

# Testing

- [ ] Structure friendly-fire code
  - Make some mages
  - Go to fuling village
  - Let mages impact the fuling structures via their AoE spells
  - The fuling structures should blow up and be destroyed
  - No errors should appear in code
- [ ] Ships ignore minion impact damage
  - Make a ship eg. Karve
  - Teleport heaps of minions onto it and/or create more
  - As you sail around, the minions shouldn't cause any damage to the ship
- [ ] No duplicate resources on death
  - In the previous versions it could happen that with `Everything` or `JustResources`, the crate would spawn with items and then the corpse would also drop those same items again. That should be fixed now
  - To test, make minions and kill them. They should only drop what went into making them ONCE (not twice)
  - **Known problem:** This doesn't fix the behavior of `Everything` and `JustResources` dropping more than they should. I'm trying to fix that as well but it's not ready yet
